addons/resource_mail/static/tests/many2many_avatar_resource.test.js

```diff
@@ -0,0 +1,194 @@
+            - a card popover should open
+            - No possibility to send a message to this resource as no user exists for it
+        - human resource linked to a user (resourcePierre)
+            - avatar of the user should be displayed
+            - a card popover should open
+            - A button to send a message to this user should be on the card
+    */
+    const pyEnv = await startServer();
+
+    // User
+    data.partnerPierreId = pyEnv["res.partner"].create({
+        name: "Pierre",
+        email: "Pierre@odoo.test",
+        phone: "+32487898933",
+    });
+    data.userPierreId = pyEnv["res.users"].create({
+        name: "Pierre",
+        partner_id: data.partnerPierreId,
+    });
+
+    // Resources
+    [data.resourceComputerId, data.resourceMarieId, data.resourcePierreId] = pyEnv[
+        "resource.resource"
+    ].create([
+        {
+            name: "Continuity testing computer",
+            resource_type: "material",
+        },
+        {
+            name: "Marie",
+            resource_type: "user",
+        },
+        {
+            name: "Pierre",
+            resource_type: "user",
+            user_id: data.userPierreId,
+            im_status: "online",
+        },
+    ]);
+
+    // Task linked to those resources
+    data.task1Id = pyEnv["resource.task"].create({
+        display_name: "Task with three resources",
+        resource_ids: [data.resourceComputerId, data.resourceMarieId, data.resourcePierreId],
+    });
+});
+test("many2many_avatar_resource widget in form view", async () => {
+    await start();
+    await openFormView("resource.task", data.task1Id, {
+        arch: `<form string="Tasks">
+                <field name="display_name"/>
+                <field name="resource_ids" widget="many2many_avatar_resource"/>
+            </form>`,
+    });
+    await contains("img.o_m2m_avatar", { count: 2 });
+    await contains(".fa-wrench", { count: 1 });
+    // Second and third records in widget should display employee avatars
+    const avatarImages = document.querySelectorAll(
+        ".many2many_tags_avatar_field_container .o_tag img"
+    );
+    expect(avatarImages[0].getAttribute("data-src")).toBe(
+        "/web/image/resource.resource/" + data.resourceMarieId + "/avatar_128"
+    );
+    expect(avatarImages[1].getAttribute("data-src")).toBe(
+        "/web/image/resource.resource/" + data.resourcePierreId + "/avatar_128"
+    );
+    // 1. Clicking on material resource's icon
+    await click(
+        document.querySelector(".many2many_tags_avatar_field_container .o_tag i.fa-wrench")
+    );
+    await contains(".o_avatar_card", { count: 0 });
+    // 2. Clicking on human resource's avatar with no user associated
+    await click(document.querySelector(".many2many_tags_avatar_field_container .o_tag img"));
+    await contains(".o_card_user_infos span", { text: "Marie" });
+    await contains(
+        ".o_avatar_card",
```

Comment:
All the `click(document.querySelector())` could just remove the `document.querySelector()` function. `click` accepts string selector as parameter. Again this simplifies the code :)